### DISCUSSION TITLE
Configure PHP Language Server

### DISCUSSION THREAD

**Discussion by eric-simpliphy** [zed-industries/zed#14215] [Feature Requests]
How can I configure the PHP language server (intelephense)? The options are documented [here](https://github.com/bmewburn/intelephense-docs/blob/master/installation.md#configuration-options). I attempted to add this to my Zed config (based on [this documentation](https://zed.dev/docs/configuring-zed#lsp)) and no dice.

```json
  "lsp": {
    "intelephense": {
      "initialization_options": {
        "intelephense.diagnostics.undefinedProperties": false
      }
    }
  }
```

**Comment by tepavcevic** [comment]
There is an extension that enables PHP lsp, have you tried installing it first?

**Reply by eric-simpliphy** [reply]
Yes, that is what I am using. Internally it used intelephense but I'm unsure how to configure it through Zed.

**Comment by xtrasmal** [comment]
Don't.
Install the php extension and you are ready to go.

**Reply by eric-simpliphy** [reply]
I did install the PHP extension. It uses intelphense internally. So what I'm trying to do is configure said PHP extension.

**Reply by xtrasmal** [reply]
What do you want to do? I can help.

**Reply by eric-simpliphy** [reply]
I just want to configure the internally used intelphense using the configuration abilities that intelphense offers (orig post had link to config options). Specifically in the above example I am trying to disable the `undefinedProperties` warning as it's too noisy on the code base I'm working on. But I imagine there will be other configuration changes I want to make to how intelphense operates as well. The Zed doc makes reference to configuring the LSP for Rust so was wanting to do the same thing for PHP.

**Reply by xtrasmal** [reply]
```json
"lsp": {
    "intelephense": {
      "max_memory": "4096M",
      "initialization_options": {
        "enable": true,
        "completion": {
          "snippetSupport": true
        },
        "format": {
          "enable": true
        },
        "codeAction": {
          "enable": true
        },
        "diagnostics": {
          "enable": true,
        "undefinedProperties": true
        },
    }
  }
}
```
you can try this, Maybe it works. Didnt test the undefinedProperties.

**Reply by eric-simpliphy** [reply]
That's a good idea to turn the `intelephense.diagnostics.undefinedProperties` defined in the intelephense documentation into an object instead of just a key. Unfortunately it didn't seem to work either. I tried exactly what you had as well as a half dozen variations of it and nothing seemed to configure intelephense. Will maybe take a dive into the Zed code later to try to better understand how configuration is provided to intelephense.

**Reply by xtrasmal** [reply]
try to get phpactor working and you will be my hero.

**Reply by josecanciani** [reply]
Does the extension work on remote servers?

**Comment by hanoii** [comment]
I am trying to also give zed with php a try and I haven't gotten very far. I installed the extension and works initially with open files, but I can't get the project to index the folder. Is there something specific I need to configure? For now using just phpactor.

I am using Latest OSX

**Reply by xtrasmal** [reply]
@hanoii you are in luck. Phpactor is now supported.
You don't have to set anything up for php now.

Join us on Discord and I'll guide you through.

**Reply by mosenturm** [reply]
@xtrasmal Can you show me a configuration for using phpactor? On Discord or here.
Thanks!

**Reply by xtrasmal** [reply]
I've got a extended readme and examples here https://github.com/xtrasmal/zed-lsp-phpactor

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
